### PR TITLE
React Todos: Version 1

### PR DESCRIPTION
**Pull request by lily4178993**
# React Todos: Version 1

## What does this PR do? 🤔 
This PR is raised to to merge the following features to the _main_ branch.

Changes that have been made:
- [X] Create a Todos application is using [Create React App](https://create-react-app.dev/)
- [X] Set-up GitHub Actions by creating linters.yml
- [X] Set-up linters:
  * Set-up ESLint
  * Set-up Stylelint
- [X] Create Components: `TodoApp`, `Header`, `InputTodo`, `TodosList`, `TodoItem`
- [X] Add functionalities to components:
     * Implemented function for CRUD opreation
- [X] Add CSS Styles files

## How should this app be tested?
Clone this repository:
```sh
git clone https://github.com/lily4178993/react-todo-app.git
```

Checkout the branch:
```sh
cd react-todo-app
git checkout dev
```

Install all dependencies:
```sh
npm install
```
Fix JS/JSX linters errors
```sh
npx eslint "**/*.{js,jsx}" --fix
```

Open the project:
```sh
npm start
```

Please kindly review my implementation and suggest any changes and improvements.  Best regards.
Thanks for reviewing my PR 😊

### CODE REVIEW

File: src/components/InputTodo.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+
+// eslint-disable-next-line react/prop-types
```
Comment (SanaTameem) [inline src/components/InputTodo.jsx:3]: - Kindly remove the eslint-disable lines from your project as this is not a good coding practice.

Comment (SanaTameem) [inline src/components/InputTodo.jsx:3]: - Kindly remove eslint disable lines from your project as it is not a good coding practice.

File: README.md [hunk 2]
```diff
@@ -1 +1,220 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+<br>
+<hr>
+
+<!-- LIVE DEMO -->
+## <b>Live Demo 🚀</b><a name="live-demo"></a>
+
+> This app will be deployed in the near future.
+
```
Comment (SanaTameem) [inline README.md:68]: - Kindly try to add the link of live demo to your project README.md file or remove the live demo section from README.

File: README.md [hunk 3]
```diff
@@ -1 +1,220 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<br><hr>
+
+<!-- LICENCE -->
+## <b>License 📝</b><a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
```
Comment (SanaTameem) [inline README.md:219]: - Kindly try to add a correct link for your MIT file.

**Review by SanaTameem — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️
### Hi @lily4178993 , 

Good job so far! :+1: 
Here are some issues that you still need to work on to go to the next project but you are almost there! 😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Good UI.
✅ Good implementation of the project requirements.

## Required Changes ♻️
_Check the comments under the review._
- [Optional] As the project requirement is to use git-flow you have to create a feature branch and open a pull request from feature to dev branch. Kindly add a feature branch to your project and don't directly push to the dev branch.
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (lily4178993) [reply README.md:207]: Hi @SanaTameem. The link for the License file is correct.

File: src/components/TodosList.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import TodoItem from 'components/TodoItem';
+
+// eslint-disable-next-line react/prop-types
```
[the remark this answers is not in the document]
Comment (lily4178993) [reply src/components/TodosList.jsx:3]: Thank you @SanaTameem. I already solved it :)

**Comment by lily4178993** [comment]
Thank you @SanaTameem for taking the time to review my pull request.
I have implemented all the changes you mentioned above.
Please review my code again.

**Review by SanaTameem — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

## Hi @lily4178993 , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  App works as expected
:heavy_check_mark:  Clean code.
:heavy_check_mark: Great job working on the changes requested

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lily4178993** [comment]
Thanks @SanaTameem.
Thank you for taking the time to review my pull request.
Have a nice day!

**Pull request merged**